model/riscv_regs.sail

```diff
@@ -16,75 +16,16 @@ register instbits : xlenbits
 
 /* register file and accessors */
 
-register x1  : regtype
-register x2  : regtype
-register x3  : regtype
-register x4  : regtype
-register x5  : regtype
-register x6  : regtype
-register x7  : regtype
-register x8  : regtype
-register x9  : regtype
-register x10 : regtype
-register x11 : regtype
-register x12 : regtype
-register x13 : regtype
-register x14 : regtype
-register x15 : regtype
-register x16 : regtype
-register x17 : regtype
-register x18 : regtype
-register x19 : regtype
-register x20 : regtype
-register x21 : regtype
-register x22 : regtype
-register x23 : regtype
-register x24 : regtype
-register x25 : regtype
-register x26 : regtype
-register x27 : regtype
-register x28 : regtype
-register x29 : regtype
-register x30 : regtype
-register x31 : regtype
+/*
+ * I vs E populates a vector of references to either x1-x15 (E) or x1-x31 (I).
+ *
+ * This is exposed as a function because we cannot forward-declare non-function
+ * values.
+ */
+val xreg_by_index : range(1,regno_range_max) -> register(regtype)
 
 function rX (Regno(r) : regno) -> xlenbits = {
-  let v : regtype =
-    match r {
-      0 => zero_reg,
-      1 => x1,
-      2 => x2,
-      3 => x3,
-      4 => x4,
-      5 => x5,
-      6 => x6,
-      7 => x7,
-      8 => x8,
-      9 => x9,
-      10 => x10,
-      11 => x11,
-      12 => x12,
-      13 => x13,
-      14 => x14,
-      15 => x15,
-      16 => x16,
-      17 => x17,
-      18 => x18,
-      19 => x19,
-      20 => x20,
-      21 => x21,
-      22 => x22,
-      23 => x23,
-      24 => x24,
-      25 => x25,
-      26 => x26,
-      27 => x27,
-      28 => x28,
-      29 => x29,
-      30 => x30,
-      31 => x31,
-      _  => {assert(false, "invalid register number"); zero_reg}
-    };
+  let v : regtype = if r == 0 then zero_reg else __deref(xreg_by_index(r));
```

Comment:
@Alasdair Would it be possible to have a less forbidding name for `__deref` in the Sail library?  It seems useful for non Sail-internal uses.

Reply:
Is there a reason why it's `*ref = v` but `let v = __deref(ref)`? Would `*` for both not be more natural (just as you can do `X(rd) = v` and `let v = X(rs1)` today)?

Reply:
`let v = *ref` should work, it's just syntax sugar for `__deref`.